### PR TITLE
refactor: convert api base clients to TS

### PR DESCRIPTION
**Pull request by vladfrangu**
Depends on #187

### CODE REVIEW

File: src/base/resource_client.ts [hunk 1]
```diff
@@ -16,17 +16,10 @@ const MAX_WAIT_FOR_FINISH = 999999;
...
  */
-class ResourceClient extends ApiClient {
-    /**
-     * @param {object} [options]
-     * @return {Promise<?object>}
-     * @private
-     */
-    async _get(options = {}) {
+export class ResourceClient extends ApiClient {
+    protected async _get<T extends Record<string, unknown>, R extends unknown>(options: T = {} as T): Promise<R | undefined> {
```
Comment (B4nan) [inline src/base/resource_client.ts:22]: does it actually make any difference to define `R extends unknown`? `unknown` is a top type, every type is its extension, so it feels redundant

File: src/base/resource_client.ts [hunk 2]
```diff
@@ -117,12 +99,15 @@ class ResourceClient extends ApiClient {
         } while ((shouldRepeat()));
```
Comment (B4nan) [inline src/base/resource_client.ts:99]: i know its unrelated, but :]
```suggestion
        } while (shouldRepeat());
```

**Review by B4nan — APPROVED** [review]
looking good, i am just not sure about the `extends unknown`, won't hurt anyone, but i don't think it has any value

File: src/base/resource_client.ts (hunk 1 above)
Comment (vladfrangu) [reply src/base/resource_client.ts:22]: oh yeah, good point. Can simplify it to just `R` 👍

**Review by B4nan — APPROVED** [review]

**Pull request merged**